app/controllers/data_api/v1/charges_controller.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module DataApi
+  module V1
+    class ChargesController < DataApi::BaseController
+      include PremiumFeatureOnly
+
+      def bulk_forecasted_usage_amount
+        charges_data = params[:charges] || []
+
+        if charges_data.empty?
+          render json: {error: "No charges provided"}, status: :bad_request
+          return
+        end
+
+        charge_ids = charges_data.map { |cd| cd[:charge_id] }.compact.uniq
+        charge_filter_ids = charges_data.map { |cd| cd[:charge_filter_id] }.compact.uniq.reject(&:blank?)
+
+        charges_lookup = Charge.where(id: charge_ids).index_by(&:id)
+        charge_filters_lookup = charge_filter_ids.any? ?
+                               ChargeFilter.where(id: charge_filter_ids).index_by(&:id) : {}
+
+        results = []
+        failed_charges = []
+
+        charges_data.each do |charge_data|
+          charge = charges_lookup[charge_data[:charge_id]]
+          charge_filter = charge_data[:charge_filter_id].present? ?
+                         charge_filters_lookup[charge_data[:charge_filter_id]] : nil
+          record_id = charge_data[:record_id]
+
+          unless charge
+            raise ActiveRecord::RecordNotFound, "Charge not found: #{charge_data[:charge_id]}"
+          end
+
+          if charge_data[:charge_filter_id].present? && !charge_filter
+            raise ActiveRecord::RecordNotFound, "ChargeFilter not found: #{charge_data[:charge_filter_id]}"
+          end
+
+          percentile_results = {}
+
+          [:units_conservative, :units_realistic, :units_optimistic].each do |percentile_key|
+            units = charge_data[percentile_key]
+            next unless units
+
+            result = Charges::CalculatePriceService.call(
+              units: units,
+              charge: charge,
+              charge_filter: charge_filter
+            )
+
+            if result.success?
+              suffix = percentile_key.to_s.gsub("units_", "")
+              percentile_results["charge_amount_cents_#{suffix}"] = result.charge_amount_cents * 100
+              percentile_results["subscription_amount_cents_#{suffix}"] = result.subscription_amount_cents * 100
+              percentile_results["total_amount_cents_#{suffix}"] = result.total_amount_cents * 100
+            end
+          end
```

Comment:
If we don't want to hardcode this here, we could nest it in an object and loop over it.

We'd also remove the string manipulation to get the suffix.

`charge_data` would be more like 

```rb
{
  charge_id: ...
  forecasted_units: {
    conservative: 100, 
    realistic: 120,
    optimistic: 300
  }
  ...
}
```

```suggestion
          charge_data[:forecasted_units].each do |estimate_label, units|
          
            result = Charges::CalculatePriceService.call(
              units: units,
              charge: charge,
              charge_filter: charge_filter
            )

            if result.success?
              percentile_results["charge_amount_cents_#{estimate_label}"] = result.charge_amount_cents * 100
              percentile_results["subscription_amount_cents_#{estimate_label}"] = result.subscription_amount_cents * 100
              percentile_results["total_amount_cents_#{estimate_label}"] = result.total_amount_cents * 100
            end
          end
```